content/flux/v0.x/get-started/syntax-basics.md

```diff
@@ -0,0 +1,353 @@
+
+##### Predicate functions
+Predicate functions use [predicate expressions](#predicate-expressions) to evaluate
+input and return `true` or `false`. For example:
+
+```js
+examplePredicate = (v) => v == "foo"
+
+examplePredicate(v: "foo")
+// Returns true
+
+examplePredicate(v: "bar")
+// Returns false
+```
+
+For more information about working with functions, see:
+
+- [Work with functions](/flux/v0.x/data-types/composite/function/)
+- [Define custom functions](/flux/v0.x/define-functions/)
+
+### Regular expression types
+A **regular expression** is a regular expression pattern used to evaluate strings.
+Use regular expressions in [predicate expressions](#predicate-expressions) or with
+the [`regexp` package](/flux/v0.x/stdlib/regexp/).
+
+```js
+regex = /^foo/
+
+"foo" =~ regex
+// Returns true
+
+"bar" =~ regex
+// Returns false
+```
+
+## Packages
+The [Flux standard library](/flux/v0.x/stdlib/) is organized into [packages](/flux/v0.x/spec/packages/)
+that contain functions and package-specific options.
+The [universe package](/flux/v0.x/stdlib/universe/) is loaded by default.
+To load other packages, include an import statement for each package at the
+beginning of your Flux script.
+
+```js
+import "array"
+import "math"
+import "influxdata/influxdb/sample"
+```
+
+## Examples of of basic syntax
+Now you can begin to apply these basic principles in real-world use cases such
+as creating data stream variables, custom functions, and more.
+
+- [Define data stream variables](#define-data-stream-variables)
+- [Define custom functions](#define-custom-functions)
+
+### Define data stream variables
+A common use case for variable assignments in Flux is creating variables for one
+or more input data streams.
+The following example uses [`sample.data()`](/flux/v0.x/stdlib/influxdata/influxdb/sample/data/)
+to query air sensor sample data.
+It assigns different streams of data to unique variables.
+
+```js
+import "influxdata/influxdb/sample"
+
+data = sample.data(set: "airSensor")
+    |> range(start: -15m)
+    |> filter(fn: (r) => r._measurement == "airSensors")
+
+temperature =
+  data
+    |> filter(fn: (r) => r._field == "temperature")
+
+humidity =
+  data
+    |> filter(fn: (r) => r._field == "humidity")
+```
+
+These variables can be used in other functions, such as  `join()`, while keeping
+the syntax minimal and flexible.
+
+### Define custom functions
+Create a function that returns the `N` number rows with the highest values in the `_value` column.
+Pass the input stream (`<-`) and the number of results to return (`n`) into a custom function.
+Use [`sort()`](/flux/v0.x/stdlib/universe/sort/) and [`limit()`](/flux/v0.x/stdlib/universe/limit/)
+to find the top `n` results in the data set.
+
+```js
+topN = (tables=<-, n) =>
+  tables
+    |> sort(desc: true)
+    |> limit(n: n)
+```
+
+Use the custom function `topN` and the `humidity` data stream variable defined
+above to return the top three data points in each input table (each room).
```

Comment:
did we introduce rooms elsewhere?

Reply:
It's just part of the sample data, but I'll remove it here.